src/pybind/mgr/cephadm/services/nvmeof.py

```diff
@@ -234,3 +239,22 @@ def post_remove(self, daemon: DaemonDescription, is_failed_deploy: bool) -> None
         _, _, err = self.mgr.mon_command(cmd)
         if err:
             self.mgr.log.error(f"Unable to send monitor command {cmd}, error {err}")
+
+    def get_blocking_daemon_hosts(self, service_name: str) -> List[HostSpec]:
+        # we should not deploy nvmeof daemons on hosts that have nvmeof daemons
+        # from services with a different "group" attribute (as recommended by
+        # the nvmeof team)
+        spec = cast(NvmeofServiceSpec, self.mgr.spec_store[service_name].spec)
+        nvmeof_group = cast(NvmeofServiceSpec, spec).group
```

Comment:
isn't this cast redundant?

Reply:
ah, yeah. I originally had this block minus the `spec = cast(NvmeofServiceSpec, self.mgr.spec_store[service_name].spec)` line somewhere else where we already had the spec and moved it. I'll take that cast out of the line grabbing the group